nixos/modules/services/monitoring/prometheus/exporters/nextcloud.nix

```diff
@@ -58,12 +59,13 @@ in
           --addr ${cfg.listenAddress}:${toString cfg.port} \
           --timeout ${cfg.timeout} \
           --server ${cfg.url} \
-          ${if cfg.tokenFile == "" then ''
+          ${if cfg.passwordFile != null then ''
             --username ${cfg.username} \
             --password ${escapeShellArg "@${cfg.passwordFile}"} \
-         '' else ''
+          '' else (if cfg.tokenFile != null then ''
             --auth-token ${escapeShellArg "@${cfg.tokenFile}"} \
-         ''} ${concatStringsSep " \\\n  " cfg.extraFlags}'';
+          '' else (abort "prometheus-exporter-nextcloud: provide either passwordFile or tokenFile"))} \
```

Comment:
i don't think this is the preferred way of implementing this. probably should just use [assertions](https://nixos.org/manual/nixos/stable/#sec-assertions).

Reply:
That sounds good in theory, but I had trouble applying it to a prometheus exporter: it looks like it should be possible (by mimicing the structure of `mkRemovedOptionModule`), but TBH it seems overkill/adds too much complexity for this PR and perhaps should be tackled as a separate contribution when there's more prometheus exporters that would benefit from assertions.